### PR TITLE
Add Metric API documentation and update L0_backend_identity CI test

### PR DESCRIPTION
**Pull request by rmccorm4**
Tests new Metric API example in https://github.com/triton-inference-server/identity_backend/pull/19 and adds documentation on Metric API

### CODE REVIEW

File: docs/metrics.md
Comment (GuanLuo) [inline docs/metrics.md:134]: something like "The user takes the ownership of the custom metrics created through the APIs and must manage their lifetime following the API documentation"?

File: docs/metrics.md
Comment (tanmayv25) [inline docs/metrics.md:138]: Link to the specific section in readme for the backend once you add the behavior. In the identity backend readme too, link back to the Custom Metrics here.

**Review by tanmayv25 — CHANGES_REQUESTED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Comment by rmccorm4** [comment]
Will wait for backend PR approval to merge both at same time

**Pull request merged**
